From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: gngpp <gngppz@gmail.com>
Date: Thu, 18 Sep 2026 00:00:00 +0800
Subject: [PATCH 12/12] boringssl: skip clang's C++ EH helper in the symbol
 audit

`__clang_call_terminate` is a linkonce_odr helper clang emits into any
translation unit with an exception-cleanup landing pad, so libssl carries one
whenever it is built with clang. It belongs to the compiler runtime rather
than BoringSSL, it cannot be prefixed, and being weak it merges instead of
colliding. The surrounding list already skips its siblings, `__cxa_*` and the
unwinder personalities.

Copyright (c) 2025-2026 0x676e67 <gngppz@gmail.com>

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
---
 util/audit_symbols.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/util/audit_symbols.go b/util/audit_symbols.go
--- a/util/audit_symbols.go
+++ b/util/audit_symbols.go
@@ -59,6 +59,7 @@ var skipWeakSymbols = []*regexp.Regexp{
 	regexp.MustCompile(`^_Zn.*`),                         // operator new()
 	regexp.MustCompile(`^__\w+\.get_pc_thunk\..*`),       // PIC
 	regexp.MustCompile(`^___asan_.*`),                    // AddressSanitizer
+	regexp.MustCompile(`^__clang_call_terminate$`),       // clang C++ EH helper
 	regexp.MustCompile(`^__cxa_.*`),                      // libc++abi
 	regexp.MustCompile(`^__dynamic_cast$`),               // libc++abi
 	regexp.MustCompile(`^__emutls_get_address$`),         // emulated TLS
